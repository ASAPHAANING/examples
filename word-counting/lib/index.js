const fs = require('fs');
const path = require('path');
const bench = require('./bench');
const js = require('./search');
const neon = require('../native/index.node');

const wc = {
  js,
  neon
};

const ROOT = path.resolve(__dirname, '..');
const DATA = path.resolve(ROOT, 'data');

const string = fs.readFileSync(
  path.resolve(DATA, 'shakespeare-plays.csv'),
  'utf8'
);
const buffer = fs.readFileSync(path.resolve(DATA, 'shakespeare-plays.csv'));

console.log(
  'Node:    ',
  bench(() => wc.js.search(string, 'thee'))
);
console.log(
  'Neon:          ',
  bench(() => wc.neon.search(buffer, 'thee'))
);
console.log(
  'Neon(parallel):',
  bench(() => wc.neon.search(buffer, 'thee', true))
);

module.exports = neon;
