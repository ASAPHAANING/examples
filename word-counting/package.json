{
  "name": "word-counting-demo",
  "description": "A word count demo to showcase Neon.",
  "main": "lib/index.js",
  "dependencies": {
    "neon-cli": "^0.2.0"
  },
  "scripts": {
    "install": "neon build --release"
  }
}
