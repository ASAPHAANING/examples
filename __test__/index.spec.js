const helloWorld = require('../hello-world');
const threadCount = require('../thread-count');
const primitives = require('../primitives');
const arrays = require('../arrays');
const objects = require('../objects');
const args = require('../arguments');
const functions = require('../functions');
const fibonacciTask = require('../fibonacci-task');
const sharingBinaryData = require('../sharing-binary-data');
const json = require('../json');
const classes = require('../classes');
const modules = require('../modules');
const errors = require('../errors');
const async = require('../async');
const {
  parse,
  stringify,
  performAsyncTask: performAsyncTaskCB
} = require('../event-emitter');
require('../word-counting');

describe('tests', () => {
  it('should run hello world', () => {
    expect(helloWorld.helloWorld()).toMatchSnapshot();
  });

  it('should run thread count', () => {
    expect(Number.isInteger(threadCount.threadCount())).toEqual(true);
  });

  it('should run primitives', () => {
    expect(primitives.primitives()).toMatchSnapshot();
  });

  it('should run arrays', () => {
    expect(arrays.convertVecToArray()).toMatchSnapshot();
    expect(arrays.returnJsArray()).toMatchSnapshot();
    expect(arrays.returnJsArrayWithNumber()).toMatchSnapshot();
    expect(arrays.returnJsArrayWithString()).toMatchSnapshot();
  });

  it('should run objects', () => {
    expect(objects.convertStructToJsObject()).toMatchSnapshot();
  });

  it('should run arguments', () => {
    expect(args.printFunction(() => {})).toMatchSnapshot();
    expect(args.add1ToArgument(1)).toMatchSnapshot();
    expect(args.getArgsLen(1, 2, 3)).toMatchSnapshot();
    expect(args.argsOpt(1)).toMatchSnapshot();
    expect(args.defaultArgs()).toMatchSnapshot();
  });

  it('should run functions', () => {
    expect(functions.requireObjectThis()).toMatchSnapshot();
    expect(functions.callJsFunction(() => 12)).toMatchSnapshot();
    expect(functions.constructJsFunction(Date)).toMatchSnapshot();
    expect(functions.returnJsFunction()).toMatchSnapshot();
  });

  it('should run json', () => {
    expect(json.sayHello('should run john')).toMatchSnapshot();
  });

  it('should run share binary data', () => {
    expect(new Float32Array(sharingBinaryData.hello())).toMatchSnapshot();
  });

  it('should run classes', () => {
    const { User } = classes;
    const user = new User(0, 'John', 'Doe', 'johndoe@gmail.com');
    expect(user).toMatchSnapshot();
  });

  it('should run fibonacci-task', async () => {
    const fib10 = await new Promise(resolve => {
      fibonacciTask.fibonacci(10, (err, result) => resolve(result));
    });
    expect(fib10).toMatchSnapshot();
    expect(fibonacciTask.fibonacciSync(10)).toMatchSnapshot();
  });

  it('should run modules', () => {
    expect(modules.hello).toMatchSnapshot();
    expect(modules.User).toMatchSnapshot();
    expect(modules.baz).toMatchSnapshot();
  });

  it('should run errors', () => {
    expect(errors.throwError).toMatchSnapshot();
    expect(errors.throwCustomError).toMatchSnapshot();
  });

  it('should run async', async () => {
    const result = await new Promise(resolve => {
      async.performAsyncTask((err, value) => resolve(value));
    });
    expect(result).toMatchSnapshot();
  });

  it('should run event emitter', async () => {
    const performAsyncTask = () =>
      new Promise(resolve => {
        performAsyncTaskCB((err, result) => {
          resolve(result);
        });
      });

    const fixture = {
      a: 1,
      b: {
        c: [
          2,
          3,
          {
            d: '4'
          }
        ]
      }
    };

    expect(parse(JSON.stringify(fixture))).toMatchSnapshot();
    expect(JSON.parse(stringify(fixture))).toMatchSnapshot();
    expect(parse(stringify(fixture))).toMatchSnapshot();
    const result = await performAsyncTask();
    expect(result).toEqual(17);
  });
});
