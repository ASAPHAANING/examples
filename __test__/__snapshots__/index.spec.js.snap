// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests arguments 1`] = `[Function]`;

exports[`tests arguments 2`] = `2`;

exports[`tests arguments 3`] = `3`;

exports[`tests arguments 4`] = `undefined`;

exports[`tests arguments 5`] = `undefined`;

exports[`tests arrays 1`] = `Array []`;

exports[`tests arrays 2`] = `Array []`;

exports[`tests arrays 3`] = `
Array [
  9000,
]
`;

exports[`tests arrays 4`] = `
Array [
  "hello node",
]
`;

exports[`tests classes 1`] = `User {}`;

exports[`tests functions 1`] = `undefined`;

exports[`tests functions 2`] = `12`;

exports[`tests functions 3`] = `1970`;

exports[`tests functions 4`] = `[Function]`;

exports[`tests json 1`] = `"Hello, john!"`;

exports[`tests objects 1`] = `
Object {
  "myNumberProperty": 1234,
  "myStringProperty": "baz",
}
`;

exports[`tests share binary data 1`] = `
Float32Array [
  0,
  0,
]
`;

exports[`tests should run hello world 1`] = `12`;

exports[`tests should run primitives 1`] = `undefined`;
